packages/gatsby-cli/src/init-starter.js

```diff
@@ -29,6 +29,20 @@ const shouldUseYarn = () => {
 
 // Initialize newly cloned directory as a git repo
 const gitInit = async rootPath => {
+  let curDir = sysPath.normalize(rootPath)
```

Comment:
I like @muescha's response & also what @sidharthachatterjee had to say

I'm thinking of something like this
```js
const isAlreadyGitRepository = async () => {
  try {
    await spawn(`git rev-parse --is-inside-work-tree`)
    return true;
  } catch (err) {
    return false
  }
}

// Clones starter from URI.
const clone = async (hostInfo: any, rootPath: string) => {
  let url
  // Let people use private repos accessed over SSH.
  if (hostInfo.getDefaultRepresentation() === `sshurl`) {
    url = hostInfo.ssh({ noCommittish: true })
    // Otherwise default to normal git syntax.
  } else {
    url = hostInfo.https({ noCommittish: true, noGitPlus: true })
  }

  const branch = hostInfo.committish ? `-b ${hostInfo.committish}` : ``

  report.info(`Creating new site from git: ${url}`)

  await spawn(`git clone ${branch} ${url} ${rootPath} --single-branch`)

  report.success(`Created starter directory layout`)

  await fs.remove(sysPath.join(rootPath, `.git`))

  const isGit = await isAlreadyGitRepository()
  await install(rootPath)
  if (!isGit) {
    await gitInit(rootPath)
  }
  await maybeCreateGitIgnore(rootPath)
  if (!isGit) {
    await createInitialGitCommit(rootPath, url)
  }
}
```